istio/kfserving-gateway/base/service.yaml

```diff
@@ -0,0 +1,50 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: kfserving-ingressgateway
+  labels:
+    app: kfserving-ingressgateway
+    kfserving: ingressgateway
+spec:
+  type: LoadBalancer
+  selector:
+    app: kfserving-ingressgateway
+    kfserving: ingressgateway
+  ports:
+    - name: status-port
+      port: 15020
+      targetPort: 15020
+    - name: http2
+      nodePort: 32380
+      port: 80
+      targetPort: 80
+    - name: https
+      nodePort: 32390
+      port: 443
+    - name: tcp
+      nodePort: 32400
+      port: 31400
+    - name: tcp-pilot-grpc-tls
+      port: 15011
+      targetPort: 15011
+    - name: tcp-citadel-grpc-tls
+      port: 8060
+      targetPort: 8060
+    - name: tcp-dns-tls
+      port: 853
+      targetPort: 853
+    - name: https-kiali
+      port: 15029
+      targetPort: 15029
+    - name: http2-prometheus
+      port: 15030
+      targetPort: 15030
+    - name: http2-grafana
+      port: 15031
```

Comment:
are these ports for kiali/grafana/prometheus always constant? do we deploy them with kubeflow?

Reply:
Thaks - great to know. What's the total infra cost of these 3 services - believe a container each? Are any customized Kubeflow metrics configured to flow to them

Reply:
@animeshsingh These require a container each. Prometheus, Graphana etc do come with the default istio install in Kubeflow.

Kubeflow does report Prometheus metrics through various operators or services like tfjob-operator or profile-controller. Likewise, the gateway container istio-proxy also reports metrics.

These ports in question are ports which the istio-proxy container exposes. The container istio-proxy is a part of the deployment servicing kfserving-gateway. These ports host https end-points for each of these services like Prometheus to poll these (gateway) services for metrics.